chia/wallet/wallet_state_manager.py

```diff
@@ -1417,17 +1399,32 @@ async def _add_coin_states(
                             additions = [state.coin for state in children]
                             if len(children) > 0:
                                 fee = 0
-
+                                known_puzzle_hash: bool = await self.puzzle_store.puzzle_hash_exists(
+                                    coin_state.coin.puzzle_hash
+                                )
                                 to_puzzle_hash = None
+                                coin_spend: Optional[CoinSpend] = None
                                 # Find coin that doesn't belong to us
                                 amount = 0
-                                for coin in additions:
+                                for child_state in children:
```

Comment:
```suggestion
                                for child_state in children:
```

Why did you change this loop to use `children` is you still only access the `coin` which is what `additions` already contains?